src/uu/seq/src/seq.rs

```diff
@@ -102,24 +130,28 @@ impl FromStr for Number {
             s = &s[1..];
         }
 
+        // If `s` is '-0', then `parse()` returns `BigInt::zero()`, but
+        // we need to return `Number::MinusZeroInt` instead.
+        //
+        // Similarly, if `s` is '-0.0', then `parse()` returns
+        // `BigDecimal::zero()`, but we need to return
+        // `Number::MinusZeroFloat` instead.
         match s.parse::<BigInt>() {
             Ok(n) => {
-                // If `s` is '-0', then `parse()` returns
-                // `BigInt::zero()`, but we need to return
-                // `Number::MinusZero` instead.
                 if n == BigInt::zero() && s.starts_with('-') {
-                    Ok(Number::MinusZero)
+                    Ok(Number::MinusZeroInt)
                 } else {
                     Ok(Number::BigInt(n))
                 }
             }
-            Err(_) => match s.parse::<f64>() {
-                Ok(value) if value.is_nan() => Err(format!(
-                    "invalid 'not-a-number' argument: '{}'\nTry '{} --help' for more information.",
-                    s,
-                    uucore::execution_phrase(),
-                )),
```

Comment:
Removing this arm of the match statement has caused a regression in the test `test_seq::test_rejects_nan`. We need to check here if the parsed value is `NaN`, but the `bigdecimal::BigDecimal` struct doesn't support NaN it seems: https://github.com/akubera/bigdecimal-rs/issues/67 . So I could add an additional `parse::<f64>()` just to check for NaN? Suggestions welcome.

Reply:
Parsing a `f64` sounds like a great solution, it seems to catch all the formats GNU seq/`strtod` catches.

You could then watch out for inf/-inf too. `seq inf` counts up endlessly, which might be useful enough to support. `seq -- -inf 0` just outputs `-inf` endlessly, that's less useful. (And `seq -- inf -1 0` outputs `inf` endlessly.)

Reply:
I'll create a new issue to represent the feature request of supporting `inf` and `-inf` arguments.

Reply:
Oh, I see what you're saying. You're saying that the `uutils` version of `seq` on the `master` branch already supports `seq inf` and `seq -- -inf 0`, and this pull request as it is today will cause those invocations to no longer work. There are no test cases for `inf` and `-inf` arguments, so I guess I need to add them now to prevent further regressions. Okay, I'll do that too.

Reply:
GNU `seq` also supports `inf`, but you're right that it would be a local regression, I didn't even think of that.